### PR TITLE
Remove legacy Authentication API endpoints

### PR DESCRIPTION
**Pull request by lbalmaceda**
### Changes
The following endpoints are considered part of the legacy authentication API and are since this PR no longer be part of the API client:

- /oauth/ro
- /oauth/access_token
- /tokeninfo
- /delegation

The associated methods that are now removed from the `AuthenticationAPIClient` class are:

- `delegation()`
- `delegationWithRefreshToken(@NonNull String refreshToken)`
- `delegationWithIdToken(@NonNull String idToken)`
- `delegationWithIdToken(@NonNull String idToken, @NonNull String apiType)`

In addition, the `AuthenticationRequest` interface no longer has a `setAccessToken(String accessToken)` method. And two constants have been removed from the `ParameterBuilder` class: `GRANT_TYPE_JWT` and `ID_TOKEN_KEY`. 

Since `Delegation` and `DelegationRequest` are no longer used, those are also removed from the codebase, along with the associated unit tests.

### References

This change was requested in SDK-2205 and SDK-2207.

### CODE REVIEW

File: V2_MIGRATION_GUIDE.md [hunk 1]
```diff
@@ -40,6 +40,8 @@ The `com.auth0.android.authentication.AuthenticationAPIClient` contains many cha
 
 If you are using the return type of any of these methods directly, you will need to change your code to expect the type as documented above. If you are chaining a call to `start` or `execute` to make the request, no changes are required.
 
+The `AuthenticationRequest` interface no longer has a `AuthenticationRequest setAccessToken(@NonNull String accessToken)` method.
```
Comment (jimmyjames) [inline V2_MIGRATION_GUIDE.md:43]: Can we provide some context for why it is removed or what to use instead (if the use case still valid)?

File: V2_MIGRATION_GUIDE.md [hunk 2]
```diff
@@ -50,6 +52,8 @@ Additionally, any classes that implemented `ParameterizableRequest` or `AuthRequ
 - The `com.auth0.android.request.ParameterizableRequest` interface has been removed. The ability to add request headers and parameters has been moved to the `com.auth0.android.request.Request` interface.
 - The `com.auth0.android.request.AuthRequest` interface has been removed. The `com.auth0.android.request.AuthenticationRequest` interface can be used instead.
 - The `com.auth0.android.provider.WebAuthActivity` class has been removed. External browser applications will always be used for authentication.
+- The `com.auth0.android.result.Delegation` class has been removed. There's no replacement for it.
+- The `com.auth0.android.authentication.request.DelegationRequest` class has been removed. There's no replacement for it.
```
Comment (jimmyjames) [inline V2_MIGRATION_GUIDE.md:56]: Similar to above, even though there is no replacement would be good to provide context for why it's been removed (e.g., uses legacy endpoints or similar).

File: V2_MIGRATION_GUIDE.md [hunk 3]
```diff
@@ -119,6 +122,11 @@ Additionally, any classes that implemented `ParameterizableRequest` or `AuthRequ
...
 - `public AuthRequest loginWithEmail(@NonNull String email, @NonNull String verificationCode, @NonNull String realmOrConnection)`. Use `public AuthenticationRequest loginWithEmail(@NonNull String email, @NonNull String verificationCode, @NonNull String realmOrConnection)` instead.
 - `public AuthRequest loginWithEmail(@NonNull String email, @NonNull String verificationCode)`. Use `public AuthenticationRequest loginWithEmail(@NonNull String email, @NonNull String verificationCode)` instead.
+- `public ParameterizableRequest<Map<String, Object>, AuthenticationException> delegation()`. There's no replacement for it.
+- `public DelegationRequest<Delegation> delegationWithIdToken(@NonNull String idToken)`. There's no replacement for it.
+- `public DelegationRequest<Delegation> delegationWithRefreshToken(@NonNull String refreshToken)`. There's no replacement for it.
+- `public DelegationRequest<Map<String, Object>> delegationWithIdToken(@NonNull String idToken, @NonNull String apiType)`. There's no replacement for it.
```
Comment (jimmyjames) [inline V2_MIGRATION_GUIDE.md:128]: Add context about why it's removed with no replacement.

File: auth0/src/main/java/com/auth0/android/authentication/AuthenticationAPIClient.java [hunk 4]
```diff
@@ -274,45 +268,6 @@ public AuthenticationRequest loginWithOTP(@NonNull String mfaToken, @NonNull Str
...
-     *          public void onFailure(AuthenticationException error) { }
-     *      });
-     * }
-     * </pre>
-     *
-     * @param token      obtained from the IdP
-     * @param connection that will be used to authenticate the user, e.g. 'facebook'
-     * @return a request to configure and start that will yield {@link Credentials}
-     */
-    @NonNull
```
Comment (jimmyjames) [inline auth0/src/main/java/com/auth0/android/authentication/AuthenticationAPIClient.java:298]: Are you planning to create a PR for v1 to deprecate this, and all other public methods removed that will be removed?

**Review by jimmyjames — CHANGES_REQUESTED** [review]
In addition to the comments in-line, let's add a topic for the removal of legacy endpoint methods in the migration guide, just to call it out at a high level.

File: auth0/src/main/java/com/auth0/android/authentication/AuthenticationAPIClient.java (hunk 4 above)
Comment (lbalmaceda) [reply auth0/src/main/java/com/auth0/android/authentication/AuthenticationAPIClient.java:298]: Yes, I'm tracking the deprecation here SDK-2206

**Review by jimmyjames — APPROVED** [review]

**Pull request merged**